server/src/server.ts

```diff
@@ -59,4 +59,9 @@ const server = app.listen(port, () => {
       next(new Error("Not Authorized"));
     }
   });
+  io.use(async (socket, next) => {
+    const userType = await getUser(socket.data.utorid);
+    socket.data["userType"] = userType.data.userType;
```

Comment:
Why not do this in the first place?
```suggestion
    socket.data.userType = userType.data.userType;
```

Reply:
I was preserving the code style from line 50, where the assignment seems to use the [""] notation.